frontend/composables/useRefBridge.ts

```diff
@@ -0,0 +1,112 @@
+import { type ModelRef } from 'vue'
+
+interface ConverterCallback<Tx, Ty> { (x: Tx) : Ty}
+
+/** This composable creates a two-way pipe between reactive arrays of 2 different types. The values circulate back
+ *  and forth transparently from A to B and B to A, the reactivity is preserved on both ends, the types are converted
+ *  when the values pass through.
+ *  It solves 3 difficulties:
+ *  1. Different types. This is the main reason you need to bridge reactive variables: values of different types
+ *     that must stay synchronized.
+ *  2. Infinite loops. When one array changes, the goal is to update the other one and you do not want that
+ *     the update triggers the first one again and so on.
+ *  3. At the moment the binding is created, it is typical that one array has initial data and the other is
+ *     empty. You do not want the empty one to erase the initial data when the binding starts.
+ * @param origRef Ref that you want to bridge with the new ref that this function will create for you.
+ * @param origToCreated (optional) Callback/Arrow function that converts an element in the original array into the type of the elements in the created array. If not provided, a basic conversion is performed, which is safe only between strings and numbers.
+ * @param createdToOrig (optional) Callback/Arrow function that converts an element in the created array into the type of the elements in the original array. If not provided, a basic conversion is performed, which is safe only between strings and numbers.
+ * */
+export function useArrayRefBridge<Torig, Tcreated> (origRef: Ref<Torig[]>|ModelRef<Torig[]>, origToCreated?: ConverterCallback<Torig, Tcreated>, createdToOrig?: ConverterCallback<Tcreated, Torig>) : Ref<Tcreated[]> {
+  const createdRef = ref<Tcreated[]>()
+  let pauseBack = false
```

Comment:
I think it must be possible to remove the pause variables (and the nextTick). (this is also valid for the ObjectRefBridge)

Example how it could work: (dummy code)
```ts
Origin changes ->
newOriginValue
const newCreatedValue = origToCreated(newOriginValue)
if(JSON.stringify(oldCreadedValue) === JSON.stringify(newCreatedValue)){
return
}
const testOriginValue = createdToOrign(newCreatedValue)
if(JSON.stringify(newOriginValue) === JSON.stringify(testOriginValue)){
warn('data conversion check failed', newOriginValue, newCreatedValue, testOriginValue)
return
}
createdRef.value = newCreatedValue
```

Reply:
Unfortunately not. It is similar to my initial solution but it lead to bugs. Sometimes, what we send at one end of the bridge might be slightly changed by the conversion call-backs so the output looks different and then these `JSON.stringify(newOriginValue) === JSON.stringify(testOriginValue)` cause infinite loops.
This is too restrictive, it would make the bridge unusable for some use cases. We can make the bridge more versatile if we ignore the content of the refs.

Maybe you mean that you are worried that 
``TS
pauseBack = true
nextTick(() => { pauseBack = false })
```
might not always work.
The only source of bug here would be that the Back watcher fires two ticks later, thus escaping the pause and starting a loop.
I am not sure that this would match Vue's specifications. I think that Vue ensures that all watchers run within the same tick, so the template is rendered with consistent data at the end of the tick:
https://vuejs.org/guide/essentials/watchers#callback-flush-timing
> By default, *a watcher's callback is called* after parent component updates (if any), and *before the owner component's DOM updates*. This means if you attempt to access the owner component's own DOM inside a watcher callback, the DOM will be in a pre-update state.

Reply:
but if the return conversion returns a different result and we skip this change wouldn't that then make the data invalid

if origin != (origin -> created -> origin) then there is something wrong in the data conversion. Otherwise we would get a wrong origin if the created changes later

Reply:
What you propose probably makes sense in a context that you have in mind but that I fail to imagine.

In which context would it be good that we check that the pair of conversion functions form a bijective mapping between the two refs?

To my mind, the contract that we have with the developer using the bridge is this one:
- each time ref `A` changes, sends its data to ref `B` through a conversion function,
- each time ref `B` changes, sends its data to ref `A` through a conversion function.

This is what we promise and do.

Forcing the pair of functions to form a bijective mapping would limit the use cases. I am not sure that it would bring a benefit to add a verification of the bijection to the contract. What I am sure about is that it would annoy colleagues using our bridge in some cases that they need to tackle.

I understand that forcing the bijection would prevent bugs for distracted developers, but it would do so at the expense of use cases for developers knowing what they are doing.

Reply:
Hmm I still think the bridge should be save in both ways. And not create different outcomes when converted back and forth. 
Here is my example:
-> 1. We load data 
-> 2.  data is converted into input data
-> 3. user changes data
-> 4. data is converted to backend data
...
-> 5. data is again loaded from the backend
-> 6. data is converted to input data  ..... and this has to be the same as 3

But we can leave it for now and adapt if needed.

Reply:
Yes, we can add a boolean parameter in the future that triggers a bijectivity check for distracted programmers who need it,
and keep it optional so uses cases like the example I cited (210° == -150°).